source/reactnative/src/components/elements/media.js

```
@@ -1,21 +1,103 @@
-import React from 'react';
+/**
+ * Media Element.
+ * 
+ * Refer https://docs.microsoft.com/en-us/adaptive-cards/authoring-cards/card-schema#media
+ */
+
+import React from "react";
 import {
     View,
-    Text,
-    StyleSheet
-} from 'react-native';
+    StyleSheet,
+    Image
+} from "react-native";
+
+import * as Constants from "../../utils/constants";
+import Video from "react-native-video";
+import ElementWrapper from './element-wrapper'
 
 export class Media extends React.Component {
+    constructor(props) {
+        super(props);
+        this.payload = props.json;
+        this.sources = this.getMediaSources(this.payload.sources)
+        this.state = {
+            currentSourceIndex: 0,
+            onLoad: false,
+        }
+    }
+
+    getMediaSources = (sources) => {
+        if (this.payload.sources.length > 0) {
+            sources.forEach(source => {
+                this.addUriAttribute(source)
+            })
+            return sources;
+        } else {
+            return null;
+        }
+    }
+
+    addUriAttribute = (source) => {
+        if (source.url) {
+            source.uri = source.url
+        }
+        return source;
+    }
+
+    videoError = () => {
+        if (this.state.currentSourceIndex < (this.sources.length - 1)) {
+            this.setState({
+                currentSourceIndex: this.state.currentSourceIndex + 1
+            })
+        }
+    }
```

Lets propagate the error back to the container to display failure notice.
@BalajiR please comment.

---

Yes @pragadeeshk , already there are error callbacks at root component. Same can be utilized for this case too.
@IbrahimSulai , you can access the error prop from `context` and pass the error details.